Add device support

HRosete opened the pull request:
# Description

Added support for the following devices:

- ADF4355
- ADL5240
- AD9162 (its attributes are different than that of its same device from the same family, AD9166)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

@tfcollins

amiclaus reviewed (commented):
shouldn't we have a `test_*` file for each device added. Currently only ad9162 has one.

HRosete commented:
> shouldn't we have a `test_*` file for each device added. Currently only ad9162 has one.

I've added the tests for the 2 devices, rebased the repo, and squashed the commits. This PR is now open for review again
